graylog2-web-interface/src/stores/configurations/ConfigurationsStore.js

```diff
@@ -49,7 +56,7 @@ const ConfigurationsStore = Reflux.createStore({
   list(configType) {
     const promise = fetch('GET', this._url(`/${configType}`));
     promise.then((response) => {
-      this.configuration[configType] = response;
+      this.setConfigurationState(configType, response);
```

Comment:
I think we do not deep copying here. What we could do is creating a new object for `configuration` instead of mutating the existing one:

```suggestion
      this.configuration = { ...this.configuration, [configType]: response };
```

Reply:
Great ! much cleaner 👍